Season 03/Jungmin/Week 19/week19_programmersSnail.java

```diff
@@ -0,0 +1,58 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class week19_programmersSnail {
+
+    public static int[] dx = {1, 0, -1};    // 아래, 오른쪽, 대각선
+    public static int[] dy = {0, 1, -1};
+    public static int n = 1;
+    public static int[][] result;
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        n = Integer.parseInt(br.readLine());
+
+        result = new int[n][n];
+
+        Snail();
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("[");
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                if (result[i][j] != 0) {
+                    if (i == n-1 && j == n-1) sb.append(result[i][j]).append("]");
+                    else sb.append(result[i][j]).append(",");
+                }
+            }
+        }
+
+        System.out.println(sb);
```

Comment:
j 는 i+1개만큼 가지고있기때문에  두번째 for문은 i+1까지만 돌게 하면 좋을것같네요!
근데 그 프로그래머스에서 [ ] 이건 배열을 나타내는거라 `[` `,` `]`는 실제로 값을 넣지않아도 될걸요 ? ? 
그리구 Solution클래스 내에서 구현해야되는건데 메인에서푸신거보니 제출말고 풀어보기만 하셨나요 ?!? 


```suggestion
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < i+1 ; j++) {
                if (result[i][j] != 0) {
                    sb.append(result[i][j])
                }
            }
        }
        System.out.println(sb);
```

Reply:
아 제출할 수 있는거였나요!?!? 실행칸이 제출칸인건가유..? 그냥 실행만 해볼 수 있는 건줄 알고 따로 풀었..

Reply:
@j-miiin 
![image](https://user-images.githubusercontent.com/27190708/160747823-3d606103-d8cf-490b-b166-77d200539c90.png)

ㅋㅋㅋ 실행은 예제 테스트케이스 실행해보는거고, 제출후 채점하기 버튼이 따로있는데 그러면 공개되지않은 테케 돌려서(백준처럼) 맞았는지 틀렸는지 알려줍니다 . . .

한번 제출해보세요!